### ISSUE TITLE
Cannot read property 'split' of undefined when running generator for SAP Business One Service Layer Metadata

### ISSUE THREAD

**Issue by gregorwolf** [SAP/cloud-sdk#176]
**Describe the bug**

In the branch [collection-issue](https://github.com/gregorwolf/sap-business-one-odata-cap/tree/collection-issue) of my test project [sap-business-one-odata-cap](https://github.com/gregorwolf/sap-business-one-odata-cap) I've added "@sap-cloud-sdk/generator": "^1.21.1" and the script:

`generate-odata-client --inputDir srv/external --outputDir srv/odata-client --forceOverwrite` 

but when running this script I get the following error:

```
(node:39117) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'split' of undefined
    at /Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/service-parser.js:395:49
    at Array.map (<anonymous>)
    at navigationProperties (/Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/service-parser.js:394:49)
    at /Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/service-parser.js:329:35
    at Array.map (<anonymous>)
    at transformEntities (/Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/service-parser.js:323:21)
    at transformServiceMetadata (/Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/service-parser.js:76:19)
    at parseService (/Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/service-parser.js:46:12)
    at /Users/gwolf/Documents/Projects/cap/sap-business-one-odata-cap/node_modules/@sap-cloud-sdk/generator/dist/parser/service-parser.js:39:16
    at Array.map (<anonymous>)
(node:39117) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:39117) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**To Reproduce**
1.  checkout the branch [collection-issue](https://github.com/gregorwolf/sap-business-one-odata-cap/tree/collection-issue)
2. run `npm i`
3. run `npm run generate-odata-client`

**Expected behavior**
Also the SAP Business One Service Layer Metadata should be supported.

**Used Versions:**
 - node version v12.16.1
 - npm version 6.14.5
 - SAP Cloud SDK version 1.21.1

**Comment by gregorwolf** [comment]
As also CAP has quite some issues consuming and providing the SAP Business One OData V4 service it is quite urgent as the Cloud SDK would be my 2nd option before doing plain HTTP calls.

**Comment by marikaner** [comment]
Sure, I can let you know when the generator will be released in a canary version, but please be aware, that all the v4 related functionality is experimental at the moment.

**Comment by marikaner** [comment]
Hey @gregorwolf,

we just released version 1.22.0 of the SAP Cloud SDK and it also contains an experimental version of the generator for OData v4 (just run the command as usual). Let us know if you encounter any issues or if have feedback, we would very much appreciate this. I will go ahead and close this issue.
